agent.go

```diff
@@ -657,23 +720,18 @@ func (a *Agent) handleInbound(m *stun.Message, local *Candidate, remote net.Addr
 			a.log.Warnf("discard message from (%s), %v", remote, err)
 			return
 		}
-	default:
-		return
-	}
 
-	remoteCandidate := a.findRemoteCandidate(local.NetworkType, remote)
-	if remoteCandidate == nil {
-		a.log.Debugf("detected a new peer-reflexive candiate: %s ", remote)
-		pflxCandidate, err := handleNewPeerReflexiveCandidate(local, remote)
-		if err != nil {
-			a.log.Warn(err.Error())
+		if remoteCandidate == nil {
+			if err := a.handleNewPeerReflexiveCandidate(local, remote); err != nil {
+				a.log.Warn(err.Error())
+			}
+			return
 		}
-
-		a.addRemoteCandidate(pflxCandidate)
+	default:
 		return
 	}
-	remoteCandidate.seen(false)
 
+	remoteCandidate.seen(false)
```

Comment:
maybe this is not possible for other reasons but the code itself doesn't prevent to have a nil remoteCandidate here if the transaction is valid (as in, it was on pendingBindingRequest and the addr match

Reply:
Sean, hope you don't mind I took the liberty of refactoring this function a bit while review it, because the switch was messing me up: https://gist.github.com/hugoArregui/fbb4f020d096cedcae65cf8de1ba49ce if you think is good I can send a commit. The only functional difference I introduce (I hope!) is related to comment I did above, I'm always handling a null remoteCandidate.